agent/agent.go

```diff
@@ -896,8 +896,10 @@ func (a *Agent) runTeardownGroupCommands(ctx context.Context, tc *taskContext) {
 
 	if teardownGroup.commands != nil {
 		a.killProcs(ctx, tc, true, "teardown group commands are starting")
-
+		ctx = utility.ContextWithAttributes(ctx, tc.taskConfig.TaskAttributes())
+		ctx, span := a.tracer.Start(ctx, "teardown_group")
 		_ = a.runCommandsInBlock(ctx, tc, *teardownGroup)
+		span.End()
```

Comment:
Nit: we typically just call `defer span.End()` in the line immediately after declaring the span

Reply:
I switched it up to have the teardown group defer and the commands end right after running them, and a comment explaining why. The outcome is the parent teardown_group encapsulates the commands and the task command cleanups and setupgroup command cleanups sequentially.